x-pack/plugins/security_solution/public/common/components/exceptions/add_exception_modal/index.tsx

```diff
@@ -0,0 +1,337 @@
+      );
+    }
+  }, [
+    setShouldDisableBulkClose,
+    exceptionItemsToAdd,
+    indexPatternLoading,
+    isSignalIndexLoading,
+    indexPatterns,
+  ]);
+
+  const onCommentChange = useCallback(
+    (value: string) => {
+      setComment(value);
+    },
+    [setComment]
+  );
+
+  const onCloseAlertCheckboxChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setShouldCloseAlert(event.currentTarget.checked);
+    },
+    [setShouldCloseAlert]
+  );
+
+  const onBulkCloseAlertCheckboxChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setShouldBulkCloseAlert(event.currentTarget.checked);
+    },
+    [setShouldBulkCloseAlert]
+  );
+
+  const enrichExceptionItems = useCallback(() => {
+    let enriched: Array<ExceptionListItemSchema | CreateExceptionListItemSchema> = [];
+    enriched =
+      comment !== ''
+        ? enrichExceptionItemsWithComments(exceptionItemsToAdd, [{ comment }])
+        : exceptionItemsToAdd;
+    if (exceptionListType === 'endpoint') {
+      const osTags = alertData ? ['windows'] : ['windows', 'macos', 'linux']; // TODO: use alert data instead
+      enriched = enrichExceptionItemsWithOS(enriched, osTags);
+    }
+    return enriched;
+  }, [comment, exceptionItemsToAdd, exceptionListType, alertData]);
+
+  const onAddExceptionConfirm = useCallback(() => {
+    // TODO: if close checkbox is selected, refresh signals table
+    if (addOrUpdateExceptionItems !== null) {
+      addOrUpdateExceptionItems(enrichExceptionItems());
+    }
+  }, [addOrUpdateExceptionItems, enrichExceptionItems]);
+
+  const isSubmitButtonDisabled = useCallback(
+    () => fetchOrCreateListError || exceptionItemsToAdd.length === 0,
+    [fetchOrCreateListError, exceptionItemsToAdd]
+  );
+
+  return (
+    <EuiOverlayMask>
+      <Modal onClose={onCancel} data-test-subj="add-exception-modal">
+        <ModalHeader>
+          <EuiModalHeaderTitle>{i18n.ADD_EXCEPTION}</EuiModalHeaderTitle>
+          <ModalHeaderSubtitle className="eui-textTruncate" title={ruleName}>
+            {ruleName}
+          </ModalHeaderSubtitle>
+        </ModalHeader>
+
+        {fetchOrCreateListError === true && (
+          <EuiCallOut title={'Error'} color="danger" iconType="alert">
+            <p>{i18n.ADD_EXCEPTION_FETCH_ERROR}</p>
+          </EuiCallOut>
+        )}
+        {fetchOrCreateListError === false && isLoadingExceptionList === true && (
+          <Loader data-test-subj="loadingAddExceptionModal" size="xl" />
+        )}
+        {fetchOrCreateListError === false &&
+          !isSignalIndexLoading &&
+          !indexPatternLoading &&
+          !isLoadingExceptionList &&
+          ruleExceptionList && (
+            <>
+              <ModalBodySection className="builder-section">
+                <EuiText>{i18n.EXCEPTION_BUILDER_INFO}</EuiText>
+                <EuiSpacer />
+                <ExceptionBuilder
+                  exceptionListItems={exceptionItemsToAdd}
```

Comment:
I think I might avoid passing this in, since the `ExceptionBuilder` manages its own state, it just takes an initial list of exception list items to start with. Might be easier to discuss this one offline to see if I missed something.